### PR TITLE
C2 review

### PR DESCRIPTION
**Pull request by JSteunou**
I made 2 different commits so you can pick if you disagree:

* one to use global Promise instead of _promise
* one that "modernize" a bit the code and fix some "self" undefined

TODO:

* inspect all setTimeout, things like this seems a little buggy because never canceled https://github.com/centrifugal/centrifuge-js/blob/c2/src/centrifuge.js#L1026
* use reducers, map and forEach instead of for...in
* share common code that could be shared (lot of repetition)
* use a modern eslint configuration or your own
* more tests

### CODE REVIEW

**Comment by FZambia** [comment]
@JSteunou many thanks, all looks great - I'll try to investigate my confusion about using self in Subscription public API I commented in pr review later today and try to test it out with Centrifuge server.

Regarding to todos:
* timeout you pointed must be canceled when reply from server received, you are right, maybe extra `timeout` field in `_callbacks` map?  
* use reducers etc - if this does not break compatibility with sth then I am ok
* share common code - this always helps
* use a modern eslint configuration or your own - I used [eslint config](https://github.com/centrifugal/centrifuge-js/blob/c2/.eslintrc) from popular webpack boilerplate I found somewhere on Github, is it non actual? If yes - let's replace to modern one.
* more tests - agree, the problem here was always amount of time, at moment I am developing too many libs and things related to Centrifuge library and Centrifugo v2 so have to sacrifice something.

File: src/subscription.js [hunk 1]
```diff
@@ -187,25 +185,17 @@ export default class Subscription extends EventEmitter {
...
+      .then(result => this._centrifuge._decoder.decodeCommandResult(type, result))
+    ;
   }
 
   publish(data) {
     return this._methodCall({
       method: this._centrifuge._methodType.PUBLISH,
       params: {
-        channel: self.channel,
+        channel: this.channel,
```
[the remark this answers is not in the document]
Comment (JSteunou) [reply src/subscription.js]: wrong branch, or not up-to-date build?

[the remark this answers is not in the document]
Comment (FZambia) [reply src/subscription.js]: will check as soon as I get home - I don't remember that I changed things here recently, so it's pretty interesting what's going on here.

[the remark this answers is not in the document]
Comment (FZambia) [reply src/subscription.js]: So `self` is proxy to `Window` (https://developer.mozilla.org/en-US/docs/Web/API/Window/self) and all my test examples used channel defined on global level. OMG.

[the remark this answers is not in the document]
Comment (JSteunou) [reply src/subscription.js]: yep, that's why I use to use `that` instead of `self`

File: src/subscription.js [hunk 2]
```diff
@@ -187,25 +185,17 @@ export default class Subscription extends EventEmitter {
...
-        self._centrifuge._call(message).then(function (result) {
-          resolve(self._centrifuge._decoder.decodeCommandResult(type, result));
-        }, function (err) {
-          reject(err);
-        });
-      }, function (err) {
-        reject(err);
-      });
-    });
+    this._subscribtionPromise
```
Comment (FZambia) [inline src/subscription.js]: missing `return`

Comment (FZambia) [reply src/subscription.js]: also typo in word `subscription`

**Comment by FZambia** [comment]
Many thanks for a help!

**Pull request merged**